viewer/viewer.js

```diff
@@ -1759,7 +1759,7 @@ app.post('/notifiers/:name/test', [noCacheJson, getSettingUserCache, checkCookie
 
 // gets a user's settings
 app.get('/user/settings', [noCacheJson, recordResponseTime, getSettingUserDb, checkPermissions(['webEnabled']), setCookie], (req, res) => {
-  let settings = req.settingUser.settings || settingDefaults;
+  let settings = Object.assign(settingDefaults, req.settingUser.settings);
```

Comment:
pretty sure this modifies target, so you'll need to clone settnigDefaults.

Reply:
Oh, yeah. Plus they both have arrays, so a clone of each is required.